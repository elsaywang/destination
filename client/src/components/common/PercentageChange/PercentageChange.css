:local(.percentageChange) {
    display: flex;
}

:local(.percentage) {
    flex: 1;
    text-align: right;
}

:local(.barContainer) {
    width: 4rem;
    position: relative;
    padding: 0 1rem;
}

:local(.middleDivider) {
    position: absolute;
    left: 50%;
    width: 0;
    height: 100%;
    border-left: 1px solid gray;
}

:local(.bar) {
    height: 50%;
    top: 25%;
    position: absolute;
}

:local(.bar--positive) {
    composes: bar;
    background: green;
    left: 50%;
}

:local(.bar--negative) {
    composes: bar;
    background: red;
    right: 50%;
}
