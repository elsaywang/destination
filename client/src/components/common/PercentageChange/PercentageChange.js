import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedNumber } from 'react-intl';
import classNames from 'classnames';
import styles from './PercentageChange.css';

const maxBarWidth = 2; // Coupled to half of 4rem width of the .barContainer class.

class PercentageChange extends Component {
    /**
     * Get the width of the colored bar that visualizes the percentage change.
     * @param  {number} percentageChange       prop, used for width of colored bar
     * @param  {Number} maxPercentageMagnitude prop, used to normalize bar's width
     * @param  {maxBarWidth} maxBarWidth       constant, used as scaling factor
     * @return {string}                        width of colored bar in `rem`
     */
    getBarWidth({ percentageChange, maxPercentageMagnitude }) {
        const minWidth = 0.1 * maxPercentageMagnitude;
        const normalizedBarWidth =
            (Math.abs(percentageChange) + minWidth) / (maxPercentageMagnitude + minWidth);
        const scaledBarWidth = normalizedBarWidth * maxBarWidth;
        const barWidth = Math.round(scaledBarWidth * 100) / 100;

        return `${barWidth}rem`;
    }

    getClassVariant(percentageChange) {
        return percentageChange >= 0 ? 'percentageChange--positive' : 'percentageChange--negative';
    }

    render() {
        const { percentageChange, maxPercentageMagnitude } = this.props;
        const percentageChangeClass = classNames(
            styles.percentageChange,
            styles[this.getClassVariant(percentageChange)],
        );
        const sign = percentageChange >= 0 ? '+' : 'â€“';
        const barWidth = this.getBarWidth({
            percentageChange,
            maxPercentageMagnitude,
        });

        return (
            <div className={percentageChangeClass}>
                <FormattedNumber
                    value={Math.abs(percentageChange)}
                    style="percent"
                    minimumFractionDigits={2}>
                    {percentage => (
                        <span className={styles.percentage}>{`${sign} ${percentage}`}</span>
                    )}
                </FormattedNumber>
                <span className={styles.barContainer}>
                    <span className={styles.middleDivider} />
                    <span style={{ width: barWidth }} className={styles.bar} />
                </span>
            </div>
        );
    }
}

PercentageChange.propTypes = {
    percentageChange: PropTypes.number.isRequired,
    maxPercentageMagnitude: PropTypes.number.isRequired,
};

export default PercentageChange;
