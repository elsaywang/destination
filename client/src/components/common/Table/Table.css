:local(.tableWrapper) .spectrum-Table-row {
    cursor: auto;
    user-select: text;
}
