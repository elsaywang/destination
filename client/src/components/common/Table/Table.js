import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { TableView } from '@react/react-spectrum/TableView';
import DataSource from './DataSource';
import {
    defaultRowHeight,
    defaultHeadHeightWithoutSelection,
    defaultMaxRows,
    defaultHeadHeight,
} from '../../../constants/rows';

/**
 * Table component that wraps the react-spectrum table components
 * (`TableView` and `TableViewDataSource`). More easily accepts props
 * for items, columns, table config, and event handlers.
 */
class Table extends Component {
    constructor({ items, columns, sortSearch, onLoadMore, allowsSelection }) {
        super();

        this.state = {
            dataSource: new DataSource({
                items,
                columns,
                sortSearch,
                onLoadMore,
                allowsSelection,
            }),
        };
    }

    /**
     * Dynamically set the height of the table's container to show up to a
     * certain number of rows.
     */
    getTableHeight({
        items,
        rowHeight = defaultRowHeight,
        allowsSelection = true,
        maxRows = defaultMaxRows,
    }) {
        const headHeight = allowsSelection ? defaultHeadHeight : defaultHeadHeightWithoutSelection;
        const bodyHeight = Math.min(items.length, maxRows) * rowHeight;

        return `${headHeight + bodyHeight}px`;
    }

    shouldComponentUpdate(nextProps) {
        if (!this.props.selectedRowIndexes || !nextProps.selectedRowIndexes) {
            return true;
        }
        if (this.props.isMaxSelectedRowsReached || nextProps.isMaxSelectedRowsReached) {
            return false;
        }
        const getSelectedRowIndexes = ({ selectedRowIndexes }) =>
            JSON.stringify(selectedRowIndexes.sort());

        return getSelectedRowIndexes(nextProps) === getSelectedRowIndexes(this.props);
    }

    componentDidUpdate() {
        const { items, columns } = this.props;

        this.state.dataSource.items = items;
        this.state.dataSource.columns = columns;
        this.state.dataSource.reloadData();
    }

    render() {
        const {
            items,
            rowHeight,
            renderCell,
            onSelectionChange,
            dataTest,
            allowsSelection,
        } = this.props;

        const height = this.getTableHeight({ items, rowHeight, allowsSelection });

        return (
            <div data-test={dataTest} className="table-wrapper" style={{ height }}>
                <TableView
                    dataSource={this.state.dataSource}
                    renderCell={renderCell}
                    rowHeight={rowHeight}
                    onSelectionChange={onSelectionChange}
                    allowsSelection={allowsSelection}
                />
            </div>
        );
    }
}

Table.propTypes = {
    items: PropTypes.array.isRequired,
    columns: PropTypes.array.isRequired,
    renderCell: PropTypes.func.isRequired,
    rowHeight: PropTypes.number,
    selectedRowIndexes: PropTypes.array,
    isMaxSelectedRowsReached: PropTypes.bool,
    onSelectionChange: PropTypes.func,
    sortSearch: PropTypes.func,
    dataTest: PropTypes.string,
    allowsSelection: PropTypes.bool,
};

export default Table;
