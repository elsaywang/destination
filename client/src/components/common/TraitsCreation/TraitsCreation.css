:local(.singleCreationlinkText) {
    display: inline-flex;
    font-size: 0.875rem;
    padding-left: 0.15em;
    vertical-align: text-bottom;
}

:local(.singleCreation) {
    cursor: pointer;
    margin-left: -1.05em;
    white-space: nowrap;
}

:local(.multiCreation) {
    display: flex;
    justify-content: flex-end;
}

:local(.message) {
    display: flex;
    padding: 0.5em;
    font-weight: 800;
    white-space: nowrap;
}
