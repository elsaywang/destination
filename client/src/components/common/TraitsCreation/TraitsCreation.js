import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MultiSignalsTraitsCreation from './MultiSignalsTraitsCreation';
import SingleSignalTraitsCreation from './SingleSignalTraitsCreation';
import { createOnboardedTraitUrl, createRuleBasedTraitUrl } from '../../../utils/urls';
import { stringifySignals } from '../../../utils/stringifySignals';

class TraitsCreation extends Component {
    getCreateTraitURL() {
        return this.shouldCreateOnboardedTrait()
            ? createOnboardedTraitUrl()
            : createRuleBasedTraitUrl();
    }

    getLinkText() {
        return this.props.signalType === 'ONBOARDED'
            ? 'Create Onboarded Trait'
            : 'Create Rule-Based Trait';
    }

    getButtonText() {
        return this.props.selectedResults.length > 1
            ? 'Create Trait From Selected Signals'
            : 'Create Trait From Selected Signal';
    }

    shouldCreateOnboardedTrait() {
        // For single signal creation, check the `signalType` of the signal result.
        // For multi signal creation, check if all selected signals are onboarded and
        // come from the same data source.
        const { signalType, doAllSelectedResultsShareSameDataSource } = this.props;

        return signalType === 'ONBOARDED' || doAllSelectedResultsShareSameDataSource;
    }

    storeSessionAndNavigateToTraits = e => {
        e.preventDefault();

        const { keyValuePairs, multiCreation, selectedResults, selectedDataSourceIds } = this.props;

        const signals = multiCreation ? selectedResults : [{ keyValuePairs }];
        const signalsParams = {
            signals: stringifySignals(signals),
            ...(this.shouldCreateOnboardedTrait() && {
                source: {
                    dataSourceIds: selectedDataSourceIds,
                },
            }),
        };

        sessionStorage.setItem('signalsParams', JSON.stringify(signalsParams));

        window.location.assign(this.getCreateTraitURL());
    };

    render() {
        const {
            multiCreation,
            selectedSignals,
            canCreateTraits,
            isMaxSignalSelectionsReached,
        } = this.props;

        return multiCreation ? (
            <MultiSignalsTraitsCreation
                selectedSignals={selectedSignals}
                storeSessionAndNavigateToTraits={this.storeSessionAndNavigateToTraits}
                isMaxSignalSelectionsReached={isMaxSignalSelectionsReached}
                multiTraitCreationButtonText={this.getButtonText()}
            />
        ) : (
            <SingleSignalTraitsCreation
                traitsCreationLabelText={this.getLinkText()}
                canCreateTraits={canCreateTraits}
                storeSessionAndNavigateToTraits={this.storeSessionAndNavigateToTraits}
            />
        );
    }
}

TraitsCreation.propTypes = {
    signalType: PropTypes.string,
    keyValuePairs: PropTypes.array,
    multiCreation: PropTypes.bool,
    selectedSignals: PropTypes.shape({
        selectionMessage: PropTypes.string,
        hasSignalSelectionsTypeWarning: PropTypes.bool,
        selectedRowIndexes: PropTypes.array,
        hasOnboardedSignalSelectionsWarning: PropTypes.bool,
    }),
    selectedDataSourceIds: PropTypes.array,
    canCreateTraits: PropTypes.bool,
    isMaxSignalSelectionsReached: PropTypes.bool,
};

export default TraitsCreation;
