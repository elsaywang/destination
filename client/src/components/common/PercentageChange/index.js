import PercentageChange from './PercentageChange';

export default PercentageChange;
