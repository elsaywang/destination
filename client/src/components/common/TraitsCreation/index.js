import TraitsCreation from './TraitsCreation';

export default TraitsCreation;
