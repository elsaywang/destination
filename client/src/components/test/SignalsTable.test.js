import React from 'react';
import { shallow } from 'enzyme';
import { FormattedNumber } from 'react-intl';
import { columnKeys } from '../../constants/columns';
import SignalsTable from '../SignalsTable';
import Table from '../../components/common/Table';
import PercentageChange from '../../components/common/PercentageChange';
import TraitsPopover from '../../containers/TraitsPopover';
import TraitsCreation from '../../components/common/TraitsCreation';

describe('<SignalsTable /> component', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(
        <SignalsTable
            results={{
                list: [
                    {
                        id: 0,
                        name: 'test1',
                        source: { sourceType: 'ANALYTICS' },
                        percentageChange: 0.1234,
                    },
                    {
                        id: 1,
                        name: 'test2',
                        source: { sourceType: 'ONBOARDED' },
                        percentageChange: -0.5678,
                    },
                ],
            }}
            signalType="ALL"
            sortSearch={mockFn}
            allowsSelection
        />,
    );

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Table />', () => {
            expect(wrapper.find(Table).exists()).toBeTruthy();
        });
    });

    describe('getColumns', () => {
        const { getColumns } = wrapper.instance();
        const includesColumn = (columns, columnKey) =>
            columns.map(column => column.key).includes(columnKey);
        const {
            keyValuePairs,
            keyName,
            valueName,
            signalType,
            signalSource,
            totalCounts,
            totalEventFires,
            percentageChange,
            includedInTraits,
        } = columnKeys;

        it('should return the correct columns when the "All" signal type filter is selected', () => {
            const actualAllSignalsColumnsIncludes = columnKey =>
                includesColumn(getColumns('ALL'), columnKey);

            expect(actualAllSignalsColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualAllSignalsColumnsIncludes(signalType)).toBeTruthy();
            expect(actualAllSignalsColumnsIncludes(signalSource)).toBeTruthy();
            expect(actualAllSignalsColumnsIncludes(totalCounts)).toBeTruthy();
            expect(actualAllSignalsColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualAllSignalsColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualAllSignalsColumnsIncludes(keyName)).toBeFalsy();
            expect(actualAllSignalsColumnsIncludes(valueName)).toBeFalsy();
            expect(actualAllSignalsColumnsIncludes(totalEventFires)).toBeFalsy();
        });

        it('should return the correct columns when the "Adobe Analytics" signal type filter is selected and Advanced Search is disabled', () => {
            const actualAnalyticsColumnsIncludes = columnKey =>
                includesColumn(getColumns('ANALYTICS'), columnKey);

            expect(actualAnalyticsColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualAnalyticsColumnsIncludes(signalType)).toBeTruthy();
            expect(actualAnalyticsColumnsIncludes(signalSource)).toBeTruthy();
            expect(actualAnalyticsColumnsIncludes(totalCounts)).toBeTruthy();
            expect(actualAnalyticsColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualAnalyticsColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualAnalyticsColumnsIncludes(keyName)).toBeFalsy();
            expect(actualAnalyticsColumnsIncludes(valueName)).toBeFalsy();
            expect(actualAnalyticsColumnsIncludes(totalEventFires)).toBeFalsy();
        });

        it('should return the correct columns when the "Adobe Analytics" signal type filter is selected and Advanced Search is enabled', () => {
            const actualAdvancedAnalyticsColumnsIncludes = columnKey =>
                includesColumn(getColumns('ANALYTICS', true), columnKey);

            expect(actualAdvancedAnalyticsColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualAdvancedAnalyticsColumnsIncludes(keyName)).toBeTruthy();
            expect(actualAdvancedAnalyticsColumnsIncludes(valueName)).toBeTruthy();
            expect(actualAdvancedAnalyticsColumnsIncludes(totalEventFires)).toBeTruthy();
            expect(actualAdvancedAnalyticsColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualAdvancedAnalyticsColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualAdvancedAnalyticsColumnsIncludes(signalType)).toBeFalsy();
            expect(actualAdvancedAnalyticsColumnsIncludes(signalSource)).toBeFalsy();
            expect(actualAdvancedAnalyticsColumnsIncludes(totalCounts)).toBeFalsy();
        });

        it('should return the correct columns when the "Actionable Log Files" signal type filter is selected', () => {
            const actualActionableLogFilesColumnsIncludes = columnKey =>
                includesColumn(getColumns('ALF'), columnKey);

            expect(actualActionableLogFilesColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualActionableLogFilesColumnsIncludes(signalType)).toBeTruthy();
            expect(actualActionableLogFilesColumnsIncludes(totalCounts)).toBeTruthy();
            expect(actualActionableLogFilesColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualActionableLogFilesColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualActionableLogFilesColumnsIncludes(keyName)).toBeFalsy();
            expect(actualActionableLogFilesColumnsIncludes(valueName)).toBeFalsy();
            expect(actualActionableLogFilesColumnsIncludes(signalSource)).toBeFalsy();
            expect(actualActionableLogFilesColumnsIncludes(totalEventFires)).toBeFalsy();
        });

        it('should return the correct columns when the "General Online Data" signal type filter is selected', () => {
            const actualGeneralOnlineDataColumnsIncludes = columnKey =>
                includesColumn(getColumns('REALTIME'), columnKey);

            expect(actualGeneralOnlineDataColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualGeneralOnlineDataColumnsIncludes(signalType)).toBeTruthy();
            expect(actualGeneralOnlineDataColumnsIncludes(totalCounts)).toBeTruthy();
            expect(actualGeneralOnlineDataColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualGeneralOnlineDataColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualGeneralOnlineDataColumnsIncludes(keyName)).toBeFalsy();
            expect(actualGeneralOnlineDataColumnsIncludes(valueName)).toBeFalsy();
            expect(actualGeneralOnlineDataColumnsIncludes(signalSource)).toBeFalsy();
            expect(actualGeneralOnlineDataColumnsIncludes(totalEventFires)).toBeFalsy();
        });

        it('should return the correct columns when the "Onboarded Records" signal type filter is selected', () => {
            const actualOnboardedRecordsColumnsIncludes = columnKey =>
                includesColumn(getColumns('ONBOARDED'), columnKey);

            expect(actualOnboardedRecordsColumnsIncludes(keyValuePairs)).toBeTruthy();
            expect(actualOnboardedRecordsColumnsIncludes(signalType)).toBeTruthy();
            expect(actualOnboardedRecordsColumnsIncludes(signalSource)).toBeTruthy();
            expect(actualOnboardedRecordsColumnsIncludes(totalCounts)).toBeTruthy();
            expect(actualOnboardedRecordsColumnsIncludes(percentageChange)).toBeTruthy();
            expect(actualOnboardedRecordsColumnsIncludes(includedInTraits)).toBeTruthy();

            expect(actualOnboardedRecordsColumnsIncludes(keyName)).toBeFalsy();
            expect(actualOnboardedRecordsColumnsIncludes(valueName)).toBeFalsy();
            expect(actualOnboardedRecordsColumnsIncludes(totalEventFires)).toBeFalsy();
        });
    });

    describe('Rendering cells', () => {
        const instance = wrapper.instance();

        afterEach(() => {
            jest.restoreAllMocks();
        });

        describe('renderCell', () => {
            const verifyRenderMethodInColumn = (renderMethodName, columnKey, data) => {
                jest.spyOn(instance, renderMethodName);
                instance.renderCell({ key: columnKey }, data);
                expect(instance[renderMethodName]).toHaveBeenCalledWith(data);
            };

            it('should call `renderKeyValuePairs` for cells in the `keyValuePairs` column', () => {
                verifyRenderMethodInColumn('renderKeyValuePairs', 'keyValuePairs', []);
            });
            it('should call `renderTotalCounts` for cells in the `totalCounts` column', () => {
                verifyRenderMethodInColumn('renderTotalCounts', 'totalCounts', 123456789);
            });
            it('should call `renderPercentageChange` for cells in the `percentageChange` column', () => {
                verifyRenderMethodInColumn('renderPercentageChange', 'percentageChange', 0.1234);
            });
            it('should call `renderIncludedInTraits` for cells in the `includedInTraits` column', () => {
                verifyRenderMethodInColumn('renderIncludedInTraits', 'includedInTraits', {
                    sids: [],
                    keyValuePairs: [],
                });
            });
        });

        describe('renderKeyValuePairs', () => {
            const { renderKeyValuePairs } = wrapper.instance();

            it('should render one key-value pair as `${key}=${value}` inside a div', () => {
                const oneKeyValuePair = [{ key: 'k', value: 'v' }];

                expect(renderKeyValuePairs(oneKeyValuePair)).toMatchSnapshot();
            });
            it('should render two key-value pairs on separate lines', () => {
                const twoKeyValuePairs = [{ key: 'k1', value: 'v1' }, { key: 'k2', value: 'v2' }];

                expect(renderKeyValuePairs(twoKeyValuePairs)).toMatchSnapshot();
            });
        });

        describe('renderTotalCounts', () => {
            const { renderTotalCounts } = wrapper.instance();
            const totalCountsWrapper = shallow(<div>{renderTotalCounts(123456789)}</div>);

            it('renders <FormattedNumber />', () => {
                expect(totalCountsWrapper.find(FormattedNumber).exists()).toBeTruthy();
            });
        });

        describe('renderPercentageChange', () => {
            const { renderPercentageChange } = wrapper.instance();
            const percentageChangeWrapper = shallow(<div>{renderPercentageChange(0.1234)}</div>);

            it('renders <PercentageChange />', () => {
                expect(percentageChangeWrapper.find(PercentageChange).exists()).toBeTruthy();
            });
            it('passes `percentageChange` prop', () => {
                expect(
                    percentageChangeWrapper.find(PercentageChange).props().percentageChange,
                ).toEqual(0.1234);
            });
            it('passes the largest percentageChange magnitude from all results as `maxPercentageMagnitude` prop', () => {
                expect(
                    percentageChangeWrapper.find(PercentageChange).props().maxPercentageMagnitude,
                ).toEqual(0.5678);
            });
        });

        describe('renderIncludedInTraits', () => {
            describe('when signals are not included in traits', () => {
                const { renderIncludedInTraits } = wrapper.instance();
                const signals = [
                    { sids: [], sourceType: 'ANALYTICS' },
                    { sids: [], sourceType: 'ONBOARDED' },
                    { sids: [], sourceType: 'REALTIME' },
                    { sids: [], sourceType: 'ALF' },
                    { sids: [], sourceType: '-' },
                ];

                it('should render <TraitsCreation />', () => {
                    signals.map(data =>
                        expect(
                            shallow(<div>{renderIncludedInTraits(data)}</div>)
                                .find(TraitsCreation)
                                .exists(),
                        ).toBeTruthy(),
                    );
                });

                it('passes `categoryType` prop`', () => {
                    signals.map(data => expect(renderIncludedInTraits(data)).toMatchSnapshot());
                });
            });

            describe('when signals are included in traits', () => {
                const { renderIncludedInTraits } = wrapper.instance();
                const data = { sids: [1, 2, 3], sourceType: 'ANALYTICS' };
                const includedInTraitsWrapper = shallow(<div>{renderIncludedInTraits(data)}</div>);

                it('should render <TraitsPopover />', () => {
                    expect(includedInTraitsWrapper.find(TraitsPopover).exists()).toBeTruthy();
                });

                it('passes `sids` prop', () => {
                    expect(includedInTraitsWrapper.find(TraitsPopover).props().sids).toEqual([
                        1,
                        2,
                        3,
                    ]);
                });
            });
        });
    });
});
