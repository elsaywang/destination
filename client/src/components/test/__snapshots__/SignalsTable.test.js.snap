// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignalsTable /> component Rendering cells renderIncludedInTraits when signals are not included in traits passes \`dataType\` prop\` 1`] = `
<TraitsCreation
  dataType="REALTIME"
  keyValuePairs={undefined}
/>
`;

exports[`<SignalsTable /> component Rendering cells renderIncludedInTraits when signals are not included in traits passes \`dataType\` prop\` 2`] = `
<TraitsCreation
  dataType="ONBOARDED"
  keyValuePairs={undefined}
/>
`;

exports[`<SignalsTable /> component Rendering cells renderIncludedInTraits when signals are not included in traits passes \`dataType\` prop\` 3`] = `
<TraitsCreation
  dataType="REALTIME"
  keyValuePairs={undefined}
/>
`;

exports[`<SignalsTable /> component Rendering cells renderIncludedInTraits when signals are not included in traits passes \`dataType\` prop\` 4`] = `
<TraitsCreation
  dataType="REALTIME"
  keyValuePairs={undefined}
/>
`;

exports[`<SignalsTable /> component Rendering cells renderKeyValuePairs should render one key-value pair as \`\${key}=\${value}\` inside a div 1`] = `
<div>
  <div>
    k=v
  </div>
</div>
`;

exports[`<SignalsTable /> component Rendering cells renderKeyValuePairs should render two key-value pairs on separate lines 1`] = `
<div>
  <div>
    k1=v1
  </div>
  <div>
    k2=v2
  </div>
</div>
`;

exports[`<SignalsTable /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalsTable
    results={
      Object {
        "list": Array [
          Object {
            "id": 0,
            "name": "test1",
            "percentageChange": 0.1234,
            "source": Object {
              "sourceType": "ANALYTICS",
            },
          },
          Object {
            "id": 1,
            "name": "test2",
            "percentageChange": -0.5678,
            "source": Object {
              "sourceType": "ONBOARDED",
            },
          },
        ],
      }
    }
    signalType="all"
    sortSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "columns": Array [
        Object {
          "key": "keyValuePairs",
          "minWidth": 150,
          "sortable": true,
          "title": "Key-Value Pair",
        },
        Object {
          "key": "signalType",
          "sortable": true,
          "title": "Signal Type",
          "width": 140,
        },
        Object {
          "key": "signalSource",
          "sortable": true,
          "title": "Signal Source",
          "width": 180,
        },
        Object {
          "key": "totalCounts",
          "sortable": true,
          "title": "Total Counts",
          "width": 100,
        },
        Object {
          "key": "percentageChange",
          "sortable": true,
          "title": "Percentage Change",
          "width": 165,
        },
        Object {
          "key": "includedInTraits",
          "sortable": true,
          "title": "Included In Traits",
          "width": 150,
        },
      ],
      "dataTest": "signals-table",
      "items": Array [
        Object {
          "id": 0,
          "includedInTraits": Object {
            "keyValuePairs": undefined,
            "sids": undefined,
            "sourceType": "ANALYTICS",
          },
          "name": "test1",
          "percentageChange": 0.1234,
          "signalSource": undefined,
          "signalType": "Adobe Analytics",
          "source": Object {
            "sourceType": "ANALYTICS",
          },
        },
        Object {
          "id": 1,
          "includedInTraits": Object {
            "keyValuePairs": undefined,
            "sids": undefined,
            "sourceType": "ONBOARDED",
          },
          "name": "test2",
          "percentageChange": -0.5678,
          "signalSource": undefined,
          "signalType": "Onboarded Records",
          "source": Object {
            "sourceType": "ONBOARDED",
          },
        },
      ],
      "onSelectionChange": [Function],
      "renderCell": [Function],
      "sortSearch": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "columns": Array [
          Object {
            "key": "keyValuePairs",
            "minWidth": 150,
            "sortable": true,
            "title": "Key-Value Pair",
          },
          Object {
            "key": "signalType",
            "sortable": true,
            "title": "Signal Type",
            "width": 140,
          },
          Object {
            "key": "signalSource",
            "sortable": true,
            "title": "Signal Source",
            "width": 180,
          },
          Object {
            "key": "totalCounts",
            "sortable": true,
            "title": "Total Counts",
            "width": 100,
          },
          Object {
            "key": "percentageChange",
            "sortable": true,
            "title": "Percentage Change",
            "width": 165,
          },
          Object {
            "key": "includedInTraits",
            "sortable": true,
            "title": "Included In Traits",
            "width": 150,
          },
        ],
        "dataTest": "signals-table",
        "items": Array [
          Object {
            "id": 0,
            "includedInTraits": Object {
              "keyValuePairs": undefined,
              "sids": undefined,
              "sourceType": "ANALYTICS",
            },
            "name": "test1",
            "percentageChange": 0.1234,
            "signalSource": undefined,
            "signalType": "Adobe Analytics",
            "source": Object {
              "sourceType": "ANALYTICS",
            },
          },
          Object {
            "id": 1,
            "includedInTraits": Object {
              "keyValuePairs": undefined,
              "sids": undefined,
              "sourceType": "ONBOARDED",
            },
            "name": "test2",
            "percentageChange": -0.5678,
            "signalSource": undefined,
            "signalType": "Onboarded Records",
            "source": Object {
              "sourceType": "ONBOARDED",
            },
          },
        ],
        "onSelectionChange": [Function],
        "renderCell": [Function],
        "sortSearch": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
